soccer/gameplay/plays/testing/four_courners_pass.py

```diff
@@ -0,0 +1,201 @@
+import robocup
+import play
+import behavior
+import skills.move
+import skills.capture
+import tactics.coordinated_pass
+import constants
+import main
+import enum
+
+
+## A testing play to demonstrate our ability to pass and recieve balls
+# One robot will pursue the ball while three other robots will pass the ball amongst themselves
+# they can only pass the ball at the corners and recieve at the corners and they cannot move
+# through the center of the square
+class FourCornerPass(play.Play):
+	class State(enum.Enum):
+		## One robot goes captures the ball while the other two gets on corners
+		# pursuing robot does not move on the first instance
+		setup = 1
+
+		## The robots pass to each other and begin setting up the next pass
+		# while the chasing robot goes for the ball.
+		passing = 2
+
+	def __init__(self):
+		super().__init__(continuous=True)
+
+		# register states - they're both substates of "running"
+		self.add_state(FourCornerPass.State.setup,
+					   behavior.Behavior.State.running)
+		self.add_state(FourCornerPass.State.passing,
+					   behavior.Behavior.State.running)
+
+		self.add_transition(behavior.Behavior.State.start,
+							FourCornerPass.State.setup, lambda: True,
+							'immediately')
+		#transition when ball is captured
+		self.add_transition(
+			FourCornerPass.State.setup, FourCornerPass.State.passing, 
+			lambda: self.subbehavior_with_name('capture').state == behavior.Behavior.
+			State.completed , 'all subbehaviors completed')
+
+		#transition when ball is passed or failed to pass
+		self.add_transition(
+			FourCornerPass.State.passing, FourCornerPass.State.setup, 
+			lambda: self.subbehavior_with_name('passer').state == behavior.Behavior.State.completed
+			or self.subbehavior_with_name('passer').state == behavior.Behavior.
```

Comment:
This might look cleaner if you did

```
lambda: self.subbehavior_with_name('passer').state == behavior.Behavior.State.completed or
	    self.subbehavior_with_name('passer').state == behavior.Behavior.State.failed
````

Reply:
Really I thought the operation always went on the second line?

Reply:
It's personal preference, but I was more-so commenting on the white space. If you make the lines in the lambda line up, it's really obvious it's part of the lambda